blog-posts/sls-security-headers/sls-security-headers.md

```diff
@@ -0,0 +1,91 @@
+
+The table below shows a subset of HTTP security headers. The description mentions the common attacks that the header protects against. The recommended value column contains optimal header values to protect your users against these threats.
+
+| Name | Value | Description |
+| --- | --- | --- |
+| [Strict-Transport-Security](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Strict-Transport-Security) | `max-age=31536000` | Informs browsers that the site should only be accessed using HTTPS, and that any future attempts to access it using HTTP should automatically be converted to HTTPS. The user's inputs will never be readable by a man on the middle (except potentially the first call ever, which does not contain any sensitive information). |
+| [X-Content-Type-Options](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/X-Content-Type-Options) | `nosniff` | Prevents older browsers from sniffing the content of files sent by the server rather than trusting the Content-Type header. This prevents the detection and execution of malicious Javascript in an image uploaded by an attacker and then retrieved by a victim. |
+| [X-Frame-Options](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/X-Frame-Options) | `SAMEORIGIN` | Prevents the display of the site in an iframe, and thus protects against click-jacking. |
+| [X-XSS-Protection](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/X-XSS-Protection) | `1; mode=block` | Prevents rendering of the page if an XSS attack is detected. |
+| [Referrer-Policy](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Referrer-Policy) | `strict-origin-when-cross-origin` | Prevents sending sensitive information (origin, path and query string) to another website or when the protocol is downgraded (HTTPS → HTTP) |
+
+An [extensive list](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers#security) can be found on the Mozilla Developer Network (MDN).
+
+These headers do not impact the application itself. Thus adding these can be done at a very low cost.
+
+### Side note on CORS headers
+
+Let’s digress a little bit to mention the Cross Origin Resource Sharing (CORS) headers. As a web application developer, you certainly already have come across them. Otherwise, believe me, you will encounter them soon or later. 😉
+
+HTTP security headers differ from CORS headers because:
+- they strengthen security features of the browser (instead of weakening the same origin policy)
+- they integrate very well in your development journey 😂
+
+## Demystifying HTTP security headers on AWS
+
+AWS mentions [several methods](https://aws.amazon.com/websites/) to deploy your website. Let's focus here on S3 and CloudFront and see how to configure security headers.
+
+### S3
+
+Amazon Simple Storage Service (S3) offers the option to host static websites. However, with this method, configuring HTTPS, and all the more so HTTP security headers, is *not possible*. 😮
```

Comment:
However, with this method, configuring HTTPS and all the more HTTP security headers is *not possible*.